add crewai tool example

samos123 opened the pull request:
* renames `input` to `cell` in the python SDK.
* bump python version to 0.0.2.

samos123 committed:
add crewai tool example

samos123 committed:
fix linter

samos123 committed:
rename input -> code

samos123 committed:
update name

nstogner commented on python/sandboxai/integrations/crewai.py:
```diff
@@ -0,0 +1,46 @@
+
+class RunIPythonCell(BaseTool):
+    name: str = "Run Python code"
+    description: str = "Run python code and shell commands in an ipython cell. Shell commands should be on a new line and start with a !."
```
start with a '!'.

nstogner commented on python/sandboxai/integrations/crewai.py:
```diff
@@ -0,0 +1,46 @@
+    code: str = Field(..., description="The code to execute in the ipython cell.")
+
+
+class RunIPythonCell(BaseTool):
```
Whats the convention on these classes, to call them a verb like Run...?

PS: I think we should add a comment about expected shutdown behavior (on python shutdown)

nstogner replied:
https://github.com/crewAIInc/crewAI-tools/tree/main/crewai_tools/tools

nstogner replied:
Thinking maybe `IPythonTool` or `IPythonSandbox` or `IPythonSandboxTool`

nstogner replied:
Looks like `Tool` as a suffix is a convention

nstogner replied:
Looks like the service as a prefix is common: https://github.com/crewAIInc/crewAI-tools/tree/main/crewai_tools/tools/browserbase_load_tool

nstogner replied:
Leads me to think `SandboxIPythonTool` and `SandboxShellTool`

samos123 replied:
I took this naming scheme from their example but you're right, their own tools use a different naming scheme.

samos123 replied:
or ``SandboxedIPythonTool`

samos123 committed:
rename integrations to experimental

samos123 committed:
address comments

samos123 committed:
add code comment

samos123 committed:
remove __del__ it causes issues

Exception ignored in: <function SandboxShellTool.__del__ at 0x7c1218c4d120>
Traceback (most recent call last):
  File "/home/samos/workspace/crewai_example/.venv/lib/python3.12/site-packages/sandboxai/experimental/crewai.py", line 44, in __del__
  File "/home/samos/workspace/crewai_example/.venv/lib/python3.12/site-packages/sandboxai/sandbox.py", line 85, in delete
  File "/home/samos/workspace/crewai_example/.venv/lib/python3.12/site-packages/sandboxai/client/v1.py", line 122, in delete_sandbox
  File "/home/samos/workspace/crewai_example/.venv/lib/python3.12/site-packages/requests/sessions.py", line 671, in delete
  File "/home/samos/workspace/crewai_example/.venv/lib/python3.12/site-packages/requests/sessions.py", line 589, in request
  File "/home/samos/workspace/crewai_example/.venv/lib/python3.12/site-packages/requests/sessions.py", line 703, in send
  File "/home/samos/workspace/crewai_example/.venv/lib/python3.12/site-packages/requests/adapters.py", line 700, in send
requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=33205): Max retries exceeded with url: /v1/spaces/default/sandboxes/ke4vp43s96ssyjo4rqeo (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7c122614bfe0>: Failed to establish a new connection: [Errno 111] Connection refused'))

samos123 committed:
bump python version for release
